staging/src/k8s.io/apiserver/pkg/server/filters/with_early_late_annotations.go

```diff
@@ -84,59 +91,43 @@ func WithStartupEarlyAnnotation(handler http.Handler, hasBeenReady lifecycleEven
 	return withStartupEarlyAnnotation(handler, hasBeenReady, exemptIfHealthProbe)
 }
 
-func withShutdownLateAnnotation(handler http.Handler, shutdownInitiated lifecycleEvent, delayDuration time.Duration, shouldExemptFn shouldExemptFunc, clock clockutils.PassiveClock) http.Handler {
+func withShutdownResponseHeader(handler http.Handler, shutdownInitiated lifecycleEvent, delayDuration time.Duration, hostGetterFn func() string, clock clockutils.PassiveClock) http.Handler {
+	if delayDuration == time.Duration(0) {
+		return handler
+	}
```

Comment:
Why carve out this case?

Reply:
in OpenShift the duration is always positive, but we also don't need to carve this out, the filter is agnostic